core/src/main/scala/kafka/server/ClientMetricsManager.scala

```diff
@@ -62,11 +118,108 @@ class ClientMetricsManager {
       info(s"Metrics collection is disabled for the client: ${clientInstance.getId.toString}")
     }
 
-    clientInstance.updateLastAccessTs(Calendar.getInstance.getTime.getTime)
+    clientInstance.updateLastAccessTs(getCurrentTime)
 
     new GetTelemetrySubscriptionResponse(data)
   }
 
+  def validatePushRequest(pushTelemetryRequest: PushTelemetryRequest,
+                          clientInfo: CmClientInformation): Unit = {
+
+    def isSupportedCompressionType(id: Int) : Boolean = {
+      try {
+        CompressionType.forId(id)
+        true
+      } catch {
+        case e: IllegalArgumentException => false
```

Comment:
Any chance you might consider logging what the ID was that was invalid? Perhaps at a DEBUG or TRACE level. I don't know if I'm thinking of weird scenarios today, but imagine a native client that isn't the same endianness as the JVM and it accidentally sends the compression ID in the wrong byte order. Then the author and/or user of said client goes crazy because it isn't using the requested compression type 😱

Reply:
We do raise an exception for every failed validation in the validatePushRequest method  and logs warning message in the logs which has the clientInstanceId in it